File: BinaryGap/BinaryGap.java
class Solution {
    public int solution(int N) {
        StringBuilder numeroBinario = new StringBuilder(Integer.toBinaryString(N));
        byte tamanho = 0;
        byte inicio,fim;
        byte soma = 0;
        while (true) {
            inicio = (byte) numeroBinario.indexOf("1");
            if (inicio ==-1) break;
            numeroBinario.setCharAt(inicio,'*');
            fim = (byte) numeroBinario.indexOf("1");
            if (fim==-1) break;
            soma = (byte) (fim-inicio-1); 
            tamanho = (soma > tamanho)?soma:tamanho;            
        }
        return tamanho;
    }
}

File: BinaryGap/BinaryGap.py
def solution(N):
    # write your code in Python 3.6
    numeroBinario = str(bin(N)[2:])
    tamanho = 0
    soma = 0
    while (True):
        inicio = numeroBinario.find('1')
        if inicio == -1:
            break;
        numeroBinario = numeroBinario.replace('1', '*', 1)
        fim = numeroBinario.find('1')
        if fim == -1:
            break
        soma = fim - inicio -1 
        tamanho = soma if soma > tamanho else tamanho            
    return tamanho    
    pass
